### PR TITLE
[SU-125] Use workbench-libs notifications

### PR DESCRIPTION
**Pull request by MatthewBemis**
Email notifications were originally only implemented in rawls, but then orch, sam, and thurloe needed them too so we extracted it all out into workbench-libs for use by any service. The work got dropped and only thurloe and sam ended up using the workbench-libs implementation. Rawls should use it too.

SUE is about to take on some notification-related work, so untangling this will make things a bit easier moving forward.

I tested this in a FiaB. If you'd like to test this out, I can point you to the existing FiaB so you don't have to spin anything up.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala [hunk 1]
```diff
@@ -266,8 +267,16 @@ object Boot extends IOApp with LazyLogging {
       val requesterPaysRole = gcsConfig.getString("requesterPaysRole")
       val projectTemplate = ProjectTemplate(projectOwners, projectEditors)
 
+      val notificationPubSubDAO = new org.broadinstitute.dsde.workbench.google.HttpGooglePubSubDAO(
```
Comment (MatthewBemis) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:272]: as a followup, I'd like to eliminate the rawls PubSubDAO implementation and just use the workbench-libs implementation everywhere. that'd make this a bit cleaner. but that's a separate issue

File: core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/NotificationsApiService.scala [hunk 2]
```diff
@@ -3,8 +3,9 @@ package org.broadinstitute.dsde.rawls.webservice
 import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
 import akka.http.scaladsl.server
 import akka.http.scaladsl.server.Directives._
-import org.broadinstitute.dsde.rawls.model.Notifications.{NotificationType, WorkspaceNotificationType}
-import org.broadinstitute.dsde.rawls.model.{Notifications, WorkspaceName}
+import org.broadinstitute.dsde.rawls.model.WorkspaceName
+import org.broadinstitute.dsde.workbench.model.Notifications
+import org.broadinstitute.dsde.workbench.model.Notifications.{NotificationType, WorkspaceNotificationType, WorkspaceName => NotificationWorkspaceName}
```
Comment (MatthewBemis) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/NotificationsApiService.scala:8]: another potential followup: harmonize the rawls and workbench-libs `WorkspaceName` case classes

**Comment by MatthewBemis** [comment]
jenkins retest - an unrelated test timed out

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/NotificationDAO.scala [hunk 3]
```diff
@@ -1,32 +0,0 @@
-package org.broadinstitute.dsde.rawls.dataaccess
```
Comment (MatthewBemis) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/NotificationDAO.scala:1]: For reviewers: almost all of the changes are removing the code duplicated by Rawls and pointing to the version in workbench-libs

File: core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala [hunk 4]
```diff
@@ -45,7 +44,7 @@ object UserService {
                   adminRegisterBillingAccountId: RawlsBillingAccountName,
                   billingProfileManagerDAO: BillingProfileManagerDAO
                  )(userInfo: UserInfo)(implicit executionContext: ExecutionContext) =
-    new UserService(userInfo, dataSource, googleServicesDAO, notificationDAO, samDAO, bqServiceFactory, bigQueryCredentialJson, requesterPaysRole, dmConfig, projectTemplate, servicePerimeterService, adminRegisterBillingAccountId, billingProfileManagerDAO)
+    new UserService(userInfo, dataSource, googleServicesDAO, samDAO, bqServiceFactory, bigQueryCredentialJson, requesterPaysRole, dmConfig, projectTemplate, servicePerimeterService, adminRegisterBillingAccountId, billingProfileManagerDAO)
```
Comment (MatthewBemis) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala:47]: NotificationDAO used to be used in UserService back when Rawls handled user registrations, but it's not needed here anymore

File: core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala [hunk 5]
```diff
@@ -74,7 +73,6 @@ object UserService {
 class UserService(protected val userInfo: UserInfo,
                   val dataSource: SlickDataSource,
                   protected val gcsDAO: GoogleServicesDAO,
-                  notificationDAO: NotificationDAO,
```
Comment (MatthewBemis) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala:77]: NotificationDAO used to be used in UserService back when Rawls handled user registrations, but it's not needed here anymore

**Review by marctalbott — APPROVED** [review]
Looks great, thanks for doing this!

**Comment by MatthewBemis** [comment]
jenkins retest - there was a registration error for one user during fiab-start

**Review by nawatts — APPROVED** [review]

**Comment by MatthewBemis** [comment]
ugh - a Sam timeout during fiab-start

jenkins retest

**Comment by MatthewBemis** [comment]
jenkins retest - google issues seem to be resolved

**Pull request merged**